arm/Microsoft.ApiManagement/service/portalsettings/deploy.bicep

```diff
@@ -0,0 +1,30 @@
+@description('Required. The name of the of the Api Management service.')
+param apiManagementServiceName string
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+@description('Required. Portal setting name')
+param name string
+
+@description('Optional. Portal setting properties.')
+param properties object = {}
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource portalSetting 'Microsoft.ApiManagement/service/portalsettings@2019-12-01' = if (!empty(properties)) {
+  name: '${apiManagementServiceName}/${name}'
+  properties: properties
```

Comment:
should this not be expanded. I believe there is an `enabled` property in here

Reply:
Just thinking type safety

Reply:
What do you mean. This is the old code:
`  resource apiManagementService_signin 'portalsettings@2019-12-01' = if (!empty(portalSignIn)) {
    name: 'signin'
    properties: portalSignIn
  }
  resource apiManagementService_signup 'portalsettings@2019-12-01' = if (!empty(portalSignUp)) {
    name: 'signup'
    properties: portalSignUp
  }`

Reply:
resource apiManagementService_signin 'portalsettings@2019-12-01' = if (!empty(portalSignIn)) {
    name: 'signin'
    properties: portalSignIn
  }
  resource apiManagementService_signup 'portalsettings@2019-12-01' = if (!empty(portalSignUp)) {
    name: 'signup'
    properties: portalSignUp
  }

Reply:
Jap, but also back then it used only a `properties` object. This could be anything and does not allow us to validate it's content. 
Hence I'd suggest to use
```bicep
@description('...')
param enabled bool = ...

resource portalSetting 'Microsoft.ApiManagement/service/portalsettings@2019-12-01' = if (!empty(properties)) {
  name: '${apiManagementServiceName}/${name}'
  properties: {
     enabled: enabled
  }
}
```

Reply:
There are different policies, so we cannot use your approach. Some are more complex see https://docs.microsoft.com/en-us/azure/templates/microsoft.apimanagement/2019-12-01-preview/service/portalsettings-delegation?tabs=bicep

Reply:
Going to ignore it for now but is something we should potentially revisit